erizo/src/erizo/media/ExternalOutput.cpp

```diff
@@ -282,38 +306,77 @@ void ExternalOutput::writeVideoData(char* buf, int len) {
 
     delete payload;
 
-    this->initContext();
-    if (video_stream_ == NULL) {
-      // could not init our context yet.
-      return;
-    }
-
     if (deliver) {
       unpackagedBufferpart_ -= unpackagedSize_;
 
+      this->initContext();
+      if (video_stream_ == NULL) {
+        // could not init our context yet.
+        return;
+      }
+
       long long currentTimestamp = head->getTimestamp();  // NOLINT
       if (currentTimestamp - firstVideoTimestamp_ < 0) {
         // we wrapped.  add 2^32 to correct this.
         // We only handle a single wrap around since that's ~13 hours of recording, minimum.
         currentTimestamp += 0xFFFFFFFF;
       }
 
+      if (!SU.Slide(currentTimestamp, video_measurements.rtcp_Packets, true)) {
+        ELOG_WARN("Slide Video Failed, TS: %lld", currentTimestamp);
+        unpackagedSize_ = 0;
+        unpackagedBufferpart_ = unpackagedBuffer_;
+        return;
+      }
+      // Distance from rtp timestamp and the last rtcp timestamp
+      int delta_timestamp = currentTimestamp - video_measurements.rtcp_Packets[0].getRtpTimestamp();  // 1/90000
+      // Distance of the last rtcp packet and the first one
+      long long addend = av_rescale_q(SU.NtpToMs(video_measurements.rtcp_Packets[0].getNtpTimestampMSW(), //NOLINT
+                       video_measurements.rtcp_Packets[0].getNtpTimestampLSW()) -
+                       SU.NtpToMs(video_first_rtcp_ntp_timeMSW, video_first_rtcp_ntp_timeLSW), (AVRational){1, 1000},
```

Comment:
consider using a variable to store `SU.NtpToMs(video_first_rtcp_ntp_timeMSW, video_first_rtcp_ntp_timeLSW)`, `SU.NtpToMs(video_measurements.rtcp_Packets[0].getNtpTimestampMSW(), video_measurements.rtcp_Packets[0].getNtpTimestampLSW())` and so on. 
it will help understanding the code (you won't need to comment it), and you won't need some NOLINT statements.